provd/main.py

```diff
@@ -493,6 +507,18 @@ def remove_devices_for_deleted_tenants(self):
         for t in removed_tenants:
             yield self.delete_devices(t)
 
+    def remove_configuration_for_deleted_tenants(self):
+        app = self._prov_service.app
+        auth_client = auth.get_auth_client()
+        auth_tenants = set(
+            tenant['uuid'] for tenant in auth_client.tenants.list()['items']
+        )
```

Comment:
Could we move the tenant listing outside of this function to avoid duplicate call for
- `remove_devices_for_deleted_tenants`
- `remove_configuration_for_deleted_tenants`